tftrt/examples/image-classification/image_classification.py

```diff
@@ -127,28 +141,57 @@ def eval_input_fn():
                 dtype=np.int32)
             labels = tf.identity(tf.constant(labels))
         else:
-            dataset = tf.data.TFRecordDataset(data_files)
-            dataset = dataset.apply(tf.contrib.data.map_and_batch(map_func=preprocess_fn, batch_size=batch_size, num_parallel_calls=8))
-            dataset = dataset.prefetch(buffer_size=tf.contrib.data.AUTOTUNE)
-            dataset = dataset.repeat(count=1)
-            iterator = dataset.make_one_shot_iterator()
-            features, labels = iterator.get_next()
+            if mode == 'validation':
+                dataset = tf.data.TFRecordDataset(data_files)
+                dataset = dataset.apply(tf.contrib.data.map_and_batch(map_func=preprocess_fn, batch_size=batch_size, num_parallel_calls=8))
+                dataset = dataset.prefetch(buffer_size=tf.contrib.data.AUTOTUNE)
+                dataset = dataset.repeat(count=1)
+                iterator = dataset.make_one_shot_iterator()
+                features, labels = iterator.get_next()
+            elif mode == 'benchmark':
+                dataset = tf.data.Dataset.from_tensor_slices(data_files)
+                dataset = dataset.apply(tf.contrib.data.map_and_batch(map_func=preprocess_fn, batch_size=batch_size, num_parallel_calls=8))
+                dataset = dataset.repeat(count=1)
+                iterator = dataset.make_one_shot_iterator()
+                features = iterator.get_next()
+                labels = np.random.randint(
+                    low=0,
+                    high=get_netdef(model).get_num_classes(),
+                    size=(batch_size),
+                    dtype=np.int32)
+                labels = tf.identity(tf.constant(labels))
+            else:
+                raise ValueError("Mode must be equal 'validation' or 'benchmark'") 
         return features, labels
 
     # Evaluate model
+    if mode == 'validation':
+        num_records = get_tfrecords_count(data_files)
+    elif mode == 'benchmark':
+        num_records = len(data_files) 
+    else:
+        raise ValueError("Mode must be equal 'validation' or 'benchmark'")
     logger = LoggerHook(
         display_every=display_every,
         batch_size=batch_size,
-        num_records=get_tfrecords_count(data_files))
+        num_records=num_records)
     tf_config = tf.ConfigProto()
     tf_config.gpu_options.allow_growth = True
     estimator = tf.estimator.Estimator(
         model_fn=model_fn,
         config=tf.estimator.RunConfig(session_config=tf_config),
         model_dir='model_dir')
-    duration_hook = DurationHook(target_duration)
-    results = estimator.evaluate(eval_input_fn, steps=num_iterations, hooks=[logger, duration_hook])
-
+    results = {}
+    if mode == 'validation':
+        results = estimator.evaluate(input_fn, steps=num_iterations, hooks=[logger])
+    elif mode == 'benchmark':
+        benchmark_hook = BenchmarkHook(target_duration, num_iterations)
```

Comment:
Could you change it to this:
`BenchmarkHook(target_duration=target_duration, iteration_limit=num_iterations)`